src/main/java/com/google/devtools/build/lib/remote/blobstore/CombinedDiskHttpBlobStore.java

```diff
@@ -14,136 +14,137 @@
 package com.google.devtools.build.lib.remote.blobstore;
 
 import com.google.common.base.Preconditions;
-import com.google.common.io.ByteStreams;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.MoreExecutors;
-import com.google.common.util.concurrent.SettableFuture;
 import com.google.devtools.build.lib.vfs.Path;
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.util.UUID;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+
 /**
  * A {@link SimpleBlobStore} implementation combining two blob stores. A local disk blob store and a
- * remote http blob store. If a blob isn't found in the first store, the second store is used, and
+ * remote blob store. If a blob isn't found in the first store, the second store is used, and
  * the blob added to the first. Put puts the blob on both stores.
  */
-public final class CombinedDiskHttpBlobStore extends OnDiskBlobStore {
+public final class CombinedDiskHttpBlobStore implements SimpleBlobStore {
   private static final Logger logger = Logger.getLogger(CombinedDiskHttpBlobStore.class.getName());
-  private final SimpleBlobStore bsHttp;
 
-  public CombinedDiskHttpBlobStore(Path root, SimpleBlobStore bsHttp) {
-    super(root);
-    this.bsHttp = Preconditions.checkNotNull(bsHttp);
+  private final SimpleBlobStore remoteCache;
+  private final OnDiskBlobStore diskCache;
+
+  public CombinedDiskHttpBlobStore(OnDiskBlobStore diskCache, SimpleBlobStore remoteCache) {
+    this.diskCache = Preconditions.checkNotNull(diskCache);
+    this.remoteCache = Preconditions.checkNotNull(remoteCache);
   }
 
   @Override
   public boolean containsKey(String key) {
```

Comment:
call this only contains to be in line with `get` now that you introduced `containsActionResult`?